src/components/ContactsList/index.js

```diff
@@ -0,0 +1,144 @@
+/* global cozy */
+import React, { Component } from "react";
+
+let sampleId = 0;
+const sampleContacts = [
+  {
+    _id: sampleId++,
+    name: { givenName: "Amélie", familyName: "Poulain" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Edward", familyName: "Snowden" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Aylin", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Benjamin", familyName: "André" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Bertrand", familyName: "Alevin" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "André", familyName: "Manoukian" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Erika", familyName: "Shell" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Brendan", familyName: "Abolivier" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "COZY", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "EDF", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "CAF", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Eric", familyName: "D" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  }
+];
+
+export const contactSort = (contact, comparedContact) => {
```

Comment:
This is where I think a comment would be good — the code is explicit and all, but a one line sentence can describe the 20 lines of code :)

Reply:
If I add comments, I'll add something like:
> this function sorts contacts regarding givenName and familyName

If I comment with more implementation details, it will be like
> this function sorts contacts with abc order, first with givenName (aka first name), then familyName (aka last name), and put empty string at the end

Well, this is very explicit. But a possible change will occur in the future and the PO ask us to change the sorting rules:
> As a user I want the contact list ordered with last name, then first name then empty.

And a developer will correct the `contactSort` function, but he will forget to change the comment.

6 months later, the PO changes the rules, again:
> As a user I want the empty contact on top of the contacts list, then contacts are ordered by first name, then last name.

And the developer reads the comment and understand the function sorts by first name then last name and put empty string at the end. So he takes empty line conditions and invert the return value.

BIM! A bug. Who is right? The comments? The code? Why are they desynchronized? Why should I do?

If I'll want to be more explicit to describe the 20 lines of code, I will refator condition into small named function: `givenNameFirst` `familyNameSecond`, `emptyNameAtTheEnd`. But I did not do it because I want `contactSort` to be a SRP function that takes a list of `contacts` with optional `givenName` and `familyName` and sort them. The implementation details, including the sorting rules, will be discovered when reading, but I guess if you want to change ruls, you'll rewrite the whole function instead of inverting conditions and return values.
```js
export const contactSort = (contact, comparedContact) => {
  const firstNameA =
    contact.name && contact.name.givenName
      ? contact.name.givenName.toUpperCase()
      : "";
  const firstNameB = comparedContact.name
    ? comparedContact.name.givenName.toUpperCase()
    : "";

  if (firstNameB === "") return -1;
  if (firstNameA === "") return 1;
  if (firstNameA < firstNameB) return -1;

  if (firstNameA > firstNameB) return 1;

  const lastNameA =
    contact.name && contact.name.familyName
      ? contact.name.familyName.toUpperCase()
      : "";
  const lastNameB =
    comparedContact.name && comparedContact.name.familyName
      ? comparedContact.name.familyName.toUpperCase()
      : "";
  if (lastNameA < lastNameB) return -1;
  if (lastNameA > lastNameB) return 1;
  return 0;
};
```